sashimi-webapp/src/components/editor-viewer/Content.vue

```diff
@@ -34,53 +36,35 @@ export default {
   data() {
     return {
       mdContent: '',
-      action: '',
       fileFormat: 'html',
       file: null,
-      editorCols: {
-        span_6_of_12: true,
-        span_12_of_12: false,
-        span_0_of_12: false
-      },
-      viewerCols: {
-        span_6_of_12: true,
-        span_12_of_12: false,
-        span_0_of_12: false
+      viewMode: 'split',
+      navbarInput: this.viewMode,
+      changeViewModeOnResize: function() {
+        if (window.innerWidth < 768 && this.viewMode === 'split') {
+          this.viewMode = 'editor';
+        }
       }
     };
   },
   watch: {
-    action(value) {
-      if (value === 'pages' || value === 'slides' || value === 'html') {
-        this.fileFormat = value;
-      } else if (value === 'editor' || value === 'viewer' || value === 'split') {
-        switch (value) {
-          case 'editor':
-            this.editorCols.span_12_of_12 = true;
-            this.editorCols.span_6_of_12 = false;
-            this.viewerCols.span_6_of_12 = false;
-            this.viewerCols.span_12_of_12 = false;
-            this.viewerCols.span_0_of_12 = true;
-            this.editorCols.span_0_of_12 = false;
-            break;
-          case 'viewer':
-            this.editorCols.span_12_of_12 = false;
-            this.editorCols.span_6_of_12 = false;
-            this.viewerCols.span_6_of_12 = false;
-            this.viewerCols.span_12_of_12 = true;
-            this.editorCols.span_0_of_12 = true;
-            this.viewerCols.span_0_of_12 = false;
-            break;
-          default:
-          // split screen
-            this.editorCols.span_6_of_12 = true;
-            this.viewerCols.span_6_of_12 = true;
-            this.editorCols.span_12_of_12 = false;
-            this.viewerCols.span_12_of_12 = false;
-            this.editorCols.span_0_of_12 = false;
-            this.viewerCols.span_0_of_12 = false;
-            break;
+    navbarInput(value) {
+      switch (value) {
+        case 'pages':
+        case 'slides':
```

Comment:
@jiayingy it is a good practice to explicitly mention that you are using the "fall-through" behaviour in switch case. So that other developers will not mistaken this as an potential error.